intel-sgx/ppid-retrieval-tool/Readme.md

```diff
@@ -0,0 +1,64 @@
+# PPID Retrieval Tool
+
+This tool provides functionality to retrieve the **Platform Provisioning ID (PPID)** from the **Provisioning Certification Enclave (PCE)**.
+
+## Overview
+
+The tool is inspired by Intel's **PCKRetrievalTool**, which is used to retrieve platform-specific information from the PCE. However, Intel’s tool only provides an API to get the encrypted PPID with given RSA 3072 public key.
+This tool follows similar process but uses it's own RSA key to decrypted the encrypted PPID from the PCE.
+
+## How It Works
+
+The solution involves creating two custom functions:
+
+1. **RSA Key-Pair Generation**: Generates an RSA key-pair from predefined parameters under our control.
+2. **PPID Decryption**: Decrypts the encrypted PPID obtained from the PCE.
+
+By leveraging the generated RSA key-pair, the tool decrypts the PPID as it is returned from the PCE, making it accessible in plaintext form.
+
+## Usage
+
+To use this tool:
+
+1. Ensure your environment is set up to communicate with the PCE.
+2. Run the tool with the appropriate permissions and parameters as described below to retrieve and decrypt the PPID.
+
+## Project Modules
+
+### 1. PPID Enclave
+The PPID Enclave module contains:
+- **RSA Key-Pair Generation**: Creates a pair of RSA keys for the PCE enclave.
+- **Encrypted PPID retrieval**: Uses PCE enclave functions to retrieve an encrypted PPID.
+- **PPID Decryption**: Uses the RSA keys to decrypt the PPID retrieved from the PCE.
+
+Both functionalities are encapsulated within the PPID enclave to prevent adversaries from accessing the RSA parameters or the private key used for PPID decryption.
+
+### 2. PCE Enclave
+This module handles the **encrypted PPID retrieval**. It contains:
+- A function to return the PPID in encrypted form.
+
+The compiled C code for this enclave is provided by Intel as a prebuilt `.so` file. To integrate this with our tool, we supply an enclave definition file (`.edl`) and generate a C wrapper using the **sgx_edger8r** tool.
```

Comment:
Can you add a link where the `pce.edl` and `libsgx_pce.signed.so.1.25.100.1` were found?